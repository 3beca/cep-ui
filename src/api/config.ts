export const BASE_URL = process.env.REACT_APP_BASE_URL || 'https://cep.tribeca.ovh';
export const EVENT_TYPE_URL = '/admin/event-types';