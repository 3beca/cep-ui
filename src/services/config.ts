export const BASE_URL = window._env_.BASE_URL;
export const EVENT_TYPE_URL = '/admin/event-types';