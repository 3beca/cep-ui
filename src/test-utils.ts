import nock from 'nock';
import { EventTypeList, EventTypeError } from './services/event-type';
import { EVENT_TYPE_URL } from './services/config';

export const setupNock = (url: string) => {
    const server = nock(url).defaultReplyHeaders({ 'access-control-allow-origin': '*' });
    // Skip Preflight CORS OPTION request
    nock(url).intercept(/./, 'OPTIONS').reply(200, undefined, { 'access-control-allow-origin': '*' }).persist();
    return server;
}

export const generateEventTypeListWith = (many: number = 5, next = false, prev = false) => {
    const list: EventTypeList = {
        results: Array.from({length: many},
            (_, idx) => ({
                id: idx + '',
                name: 'Elemento ' + idx,
                url: 'http://cep/elemento' + idx,
                createdAt: '2020-01-01T10:10:10.123Z',
                updatedAt: '2020-01-01T10:10:10.123Z'
            })
        )
    };
    if (prev) list.prev = 'http://cep/?page=prev';
    if (next) list.next = 'http://cep/?page=next';
    return list;
};

export const serverGetEventTypeList = (server: nock.Scope, page: number = 1, size: number = 10, status: number = 200, response: EventTypeList|EventTypeError = generateEventTypeListWith(10, false, false)) => {
    return server.get(EVENT_TYPE_URL + `/?page=${page}&pageSize=${size}`).reply(status, response);
};
export const serverDeleteEventType = (server: nock.Scope, eventTypeId: string, status: number = 204, response?: undefined|EventTypeError) => {
    return server.delete(EVENT_TYPE_URL + `/${eventTypeId}`).reply(status, response);
};