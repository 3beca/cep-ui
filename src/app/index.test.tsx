import React from 'react';
import { render, waitFor } from '@testing-library/react';
import App from './index';
import {setupNock, serverGetEventTypeList} from '../test-utils';
import {BASE_URL} from '../services/config';


test('App snapshot', async () => {
    serverGetEventTypeList(setupNock(BASE_URL), 1, 10);
    const { container, getAllByLabelText } = render(<App />);
    await waitFor(() => expect(getAllByLabelText('event name')).toHaveLength(10));
    expect(container).toMatchSnapshot();
});
