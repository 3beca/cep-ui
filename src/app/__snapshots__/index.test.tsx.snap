// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-root-2 MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-icon-3 MuiSvgIcon-fontSizeLarge"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM12.5 8H11v6l4.75 2.85.75-1.23-4-2.37V8zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
          />
        </svg>
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
        >
          CEP Service
        </h6>
      </div>
    </header>
    CEP UI
    <div
      class="makeStyles-root-86"
    >
      <div
        class="makeStyles-root-87"
      >
        <span>
          Table of Event Types
        </span>
        <div
          class="MuiTableContainer-root"
        >
          <table
            class="MuiTable-root"
          >
            <thead
              class="MuiTableHead-root makeStyles-head-88"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox"
                  scope="col"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-118 MuiCheckbox-root makeStyles-mainCheck-90"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-121"
                        data-indeterminate="false"
                        type="checkbox"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                  scope="col"
                >
                  <p
                    class="MuiTypography-root makeStyles-headText-89 MuiTypography-body1"
                  >
                    Event Type Name
                  </p>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                  scope="col"
                >
                  <p
                    class="MuiTypography-root makeStyles-headText-89 MuiTypography-body1"
                  >
                    URL
                  </p>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                  scope="col"
                >
                  <p
                    class="MuiTypography-root makeStyles-headText-89 MuiTypography-body1"
                  >
                    Last updated
                  </p>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                  scope="col"
                >
                  <p
                    class="MuiTypography-root makeStyles-headText-89 MuiTypography-body1"
                  >
                    Created At
                  </p>
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-118 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-121"
                        data-indeterminate="false"
                        type="checkbox"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </td>
                <td
                  aria-label="event name"
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  Elemento 0
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  http://cep/elemento0
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-118 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-121"
                        data-indeterminate="false"
                        type="checkbox"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </td>
                <td
                  aria-label="event name"
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  Elemento 1
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  http://cep/elemento1
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-118 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-121"
                        data-indeterminate="false"
                        type="checkbox"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </td>
                <td
                  aria-label="event name"
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  Elemento 2
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  http://cep/elemento2
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-118 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-121"
                        data-indeterminate="false"
                        type="checkbox"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </td>
                <td
                  aria-label="event name"
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  Elemento 3
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  http://cep/elemento3
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-118 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-121"
                        data-indeterminate="false"
                        type="checkbox"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </td>
                <td
                  aria-label="event name"
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  Elemento 4
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  http://cep/elemento4
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-118 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-121"
                        data-indeterminate="false"
                        type="checkbox"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </td>
                <td
                  aria-label="event name"
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  Elemento 5
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  http://cep/elemento5
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-118 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-121"
                        data-indeterminate="false"
                        type="checkbox"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </td>
                <td
                  aria-label="event name"
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  Elemento 6
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  http://cep/elemento6
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-118 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-121"
                        data-indeterminate="false"
                        type="checkbox"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </td>
                <td
                  aria-label="event name"
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  Elemento 7
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  http://cep/elemento7
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-118 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-121"
                        data-indeterminate="false"
                        type="checkbox"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </td>
                <td
                  aria-label="event name"
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  Elemento 8
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  http://cep/elemento8
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-118 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-121"
                        data-indeterminate="false"
                        type="checkbox"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </td>
                <td
                  aria-label="event name"
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  Elemento 9
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  http://cep/elemento9
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  2020-1-1 11:10:10
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="MuiTablePagination-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
          >
            <div
              class="MuiTablePagination-spacer"
            />
            <p
              class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
            >
              Rows per page:
            </p>
            <div
              class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
            >
              <div
                aria-haspopup="listbox"
                aria-label="Rows per page:"
                class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
                role="button"
                tabindex="0"
              >
                10
              </div>
              <input
                type="hidden"
                value="10"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
            >
              1-10 of more than 10
            </p>
            <div
              class="MuiTablePagination-actions"
            >
              <button
                aria-label="Previous page"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                title="Previous page"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </span>
              </button>
              <button
                aria-label="Next page"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                title="Next page"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
